### ISSUE TITLE
ScatterSeries returns DataPoint on mouse down event

### ISSUE THREAD

**Issue by interdrift** [oxyplot/oxyplot#674]
If you register the mousedown event to be handled for the ScatterSeries you can retrieve a Data Point from the event after you inverse transform the Screen Point object. The problem is that the ScatterSeries works with ScatterPoints and not Data Point so the mapping between the two is missing. Shouldn't it be more convenient to contain a Scatter Point in the event OxyMouseDownEventArgs?

Example :         

```
        private void GapFillingViewModel_MouseDown(object sender, OxyMouseDownEventArgs e) {
            var scatter = sender as ScatterSeries;
            var p = (sender as ScatterSeries).InverseTransform(e.Position);
            Console.WriteLine();
        }

**Comment by objorke** [comment]
I think I agree with you. But I might have had a reason for it. Investigations needed. Probably a breaking change?

**Comment by interdrift** [comment]
Sooner or later I will take a look on this ( read next 1 month ), if anyone else is doing some work on it, please do share to avoid parallel work on same issue.

**Comment by interdrift** [comment]
I have an update on this : 

PlotModel has a method named GetSeriesFromPoint(ScreenPoint a,limit x). It returns a single Series object. This however doesn't quite fix this issue since it works on the assumption that all the points on the screen point provided will be of a single series which is a wrong assumption.

**Comment by tibel** [comment]
`InverseTransform` returns a the point in data space. Therefor `DataPoint` is used.
It has nothing to do with the type of your data.

Also I think there is an issue that `ScatterPoint` should become a struct.

`DataPoint` and `ScatterPoint` are optimized data structures used by OxyPlot. Maybe a `Tag` or `Item`property should be added to link to your custom data type.

**Comment by objorke** [comment]
Yes, we should refactor the scatter points #140

**Comment by interdrift** [comment]
@tibel indeed it does, but we lose the link between the scatter point and the data point ( and all the extra data a potential scatter point derivative might have )

**Comment by tibel** [comment]
Link between data point and scatter point? The scatter point is the data point. Or do you mean screen point?

##### To clarify my position
`Transform` and `InverseTransfrom` translate between data coordinates and screen coordinates.
The translation between the coordinate systems is done by the axis. The `Axis` does not know anything about the series (and should not). Therefor the translation returns only the coordinates not the data structure (e.g. `ScatterPoint`) that is used in the series.

Translating data coordinates to a data point is an extra step. This might include interpolation or some hit rectangle calculation (see implementation of `GetNearestPoint` method).

That's why I think the current behavior of the transform methods is correct.

@interdrift I suspect that `GetNearestPoint` is the method you are searching for. But I have to admit that it does not return `ScatterPoint` at the moment. That should be changed/fixed.

@objorke Maybe `HitTest` and `GetNearestPoint` needs some rework to support such use cases?

**Comment by objorke** [comment]
I agree that that `HitTest`, `GetNearestPoint` and `TrackerHitResult` could need some clean up :-)
I will create a new issue regarding refactoring!

**Comment by objorke** [comment]
@interdrift I think it should be possible to retrieve the `ScatterPoint` from the `Index` property of the `TrackerHitResult`: https://github.com/oxyplot/oxyplot/blob/develop/Source/OxyPlot/Series/ScatterSeries%7BT%7D.cs#L293

**Comment by interdrift** [comment]
@objorke That seems like a good place to start refactoring. What is the insight behind using a single item behind a TrackerHitResult, two points could have the same value and this would return only one of the points.

**Comment by Hamzeh-Alchami** [comment]
l;kk;kl;
